pynblint/repository.py

```diff
@@ -72,45 +74,36 @@ def large_file_paths(self) -> List[Path]:
 
     def _get_dependencies(self) -> set:
         dependencies = set()
-        for root, dirs, files in os.walk(self.path):
-            for f in files:
-                # check if exist a requiremets.txt file
-                if f.endswith("requirements.txt"):
-                    try:
-                        with open((Path(root) / Path(f)), "r") as fi:
-                            file_row = fi.read()
-                            # print(file_row)
-                            tmp = file_row.split("\n")
-                            for item in tmp:
-                                dependencies.add(item)
-                            # dependencies.add(file_row)
-                            # dependencies.add(file_row)
-                    except Exception as e:
-                        print(e)
-                elif f.endswith("poetry.lock"):
-                    continue
-                elif f.endswith("pyproject.toml"):
-                    continue
-                elif f.endswith("environment.yml"):
-                    continue
-                elif f.endswith("setup.py"):
-                    continue
-                elif f.endswith("Pipfile"):
-                    continue
+        path = os.path.dirname(os.path.abspath(__file__))
+        path = os.path.dirname(path)
+        # siccome il file di requirement si toverà nella root di
+        # progetto ho creato un file fittizio con le stesse
+        # informazioni per non spostare quello pre-esistente
+        file = "kk.txt"
+        final_path = os.path.join(path, file)
+        with open(final_path, "r") as fi:
+            file_row = fi.read()
+            tmp = file_row.split("\n")
+            for item in tmp:
+                item.strip()
+                dependencies.add(tuple(item.split("==")))
+
         # print(dependencies)
 
         return dependencies
 
+    # momentaneamente la lascio qui ma quando ma dovo la sposterò
+    # perchè ovviamente non è una funzione che riguarda il repository
     def _get_core_dependecies(self) -> set:
         coredependecies = set()
-        # path = None
-        for i in sys.path:
-            if i.endswith("\\lib\\site-packages"):
-                for root, dirs, files in os.walk(i):
-                    for f in files:
-                        if f.endswith(".py"):
-                            coredependecies.add(f)
-                break
+        modules = sys.modules
```

Comment:
Are you sure that this solution is appropriate? From [this SO thread](https://stackoverflow.com/questions/6463918/how-to-get-a-list-of-all-the-python-standard-library-modules) I learn that:

> `sys.modules` - only shows modules that have already been loaded

The solution proposed [in one of the answers](https://stackoverflow.com/a/21659703/4178082) to the same SO question seems to be a lot cleaner to me and might be the way to go: i.e., using isort's `place_module` function or getting the list of core modules like this:

```python
from isort.stdlibs.py39 import stdlib
for name in sorted(stdlib): print(name)
```